Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -415,6 +438,26 @@ NAKED void thread_context_first_enter(void)
         "b restore_context_and_eret \n");
 }
 
+NAKED void do_assume_context(Thread*, u32)
+{
+    // clang-format off
+    // FIXME: I hope (Thread* thread, InterruptsState new_interrupts_state) aren't compiled away
```

Comment:
is this saying that we hope the arguments to the function aren't deleted? I don't think the compiler can do that... though the fact that this is a naked function and we don't want a normal prologue/epilogue makes me wonder whether we should put it in an asm file.

Reply:
Yep, I also don't think it can, that comment was copied as-is from the x86 `do_assume_context`. I'll remove it.
Putting it in an asm file would be a bit annoying, because of the usages of `REGISTER_STATE_SIZE` and `TRAP_FRAME_SIZE`.